Benchmark tests are flaky

## 🐛 Bug
I have seen the benchmark testing failing due to the following error quite a number of times:

```
>       assert np.mean(diffs) < max_diff, f"Lightning diff {diffs} was worse than vanilla PT (threshold {max_diff})"
E       AssertionError: Lightning diff [1.30557255e+00 3.60853071e-07] was worse than vanilla PT (threshold 0.0002)
E       assert 0.6527864532870287 < 0.0002
E        +  where 0.6527864532870287 = <function mean at 0x7fcd121d6320>(array([1.30557255e+00, 3.60853071e-07]))
E        +    where <function mean at 0x7fcd121d6320> = np.mean

tests/benchmarks/test_basic_parity.py:38: AssertionError
```
https://dev.azure.com/PytorchLightning/pytorch-lightning/_build/results?buildId=62256&view=logs&j=46f8844e-d2e7-54ff-5b07-42679d5a20a0&t=d0c9a9e3-f716-53b7-5c53-27dd0366278d

which is configured at:

https://github.com/PyTorchLightning/pytorch-lightning/blob/bc1c8b926c5072f58f42ad4b7413a8ef5c904c85/.azure-pipelines/gpu-tests.yml#L121-L123
https://github.com/PyTorchLightning/pytorch-lightning/blob/bc1c8b926c5072f58f42ad4b7413a8ef5c904c85/.azure-pipelines/gpu-benchmark.yml#L39

### To Reproduce

### Expected behavior
No error raised so we can always keep our CI green 🟢

### Environment

### Additional context


cc @carmocca @akihironitta @borda

--- a/tests/benchmarks/test_basic_parity.py
+++ b/tests/benchmarks/test_basic_parity.py
@@ -149,7 +149,6 @@ def vanilla_loop(cls_model, idx, device_type: str = "cuda", num_epochs=10):
 
 def lightning_loop(cls_model, idx, device_type: str = "cuda", num_epochs=10):
     seed_everything(idx)
-    torch.backends.cudnn.deterministic = True
 
     model = cls_model()
     # init model parts
@@ -163,6 +162,7 @@ def lightning_loop(cls_model, idx, device_type: str = "cuda", num_epochs=10):
         devices=1,
         logger=False,
         replace_sampler_ddp=False,
+        benchmark=False,
     )
     trainer.fit(model)
 

--- a/tests/benchmarks/test_sharded_parity.py
+++ b/tests/benchmarks/test_sharded_parity.py
@@ -138,7 +138,13 @@ def plugin_parity_test(
     use_cuda = gpus > 0
 
     trainer = Trainer(
-        fast_dev_run=True, max_epochs=1, accelerator="gpu", devices=gpus, precision=precision, strategy="ddp_spawn"
+        fast_dev_run=True,
+        max_epochs=1,
+        accelerator="gpu",
+        devices=gpus,
+        precision=precision,
+        strategy="ddp_spawn",
+        benchmark=False,
     )
 
     max_memory_ddp, ddp_time = record_ddp_fit_model_stats(trainer=trainer, model=ddp_model, use_cuda=use_cuda)
@@ -154,6 +160,7 @@ def plugin_parity_test(
         devices=gpus,
         precision=precision,
         strategy="ddp_sharded_spawn",
+        benchmark=False,
     )
     assert isinstance(trainer.strategy, DDPSpawnShardedStrategy)
 

